stl/inc/stacktrace

```diff
@@ -0,0 +1,365 @@
+
+// Some of these functions may throw
+// (they would propagate bad_alloc potentially thrown from string::resize_and_overwrite)
+
+void __stdcall __std_stacktrace_address_to_string(
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+void __stdcall __std_stacktrace_description( //
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+void __stdcall __std_stacktrace_source_file( //
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+_NODISCARD unsigned int __stdcall __std_stacktrace_source_line(const void* _Address) noexcept;
+
+void __stdcall __std_stacktrace_to_string(
+    const void* _Addresses, size_t _Size, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+_NODISCARD void* __stdcall __std_stacktrace_get_debug_interface() noexcept;
+_END_EXTERN_C
+
+_STD_BEGIN
+inline size_t __stdcall _Stacktrace_string_fill_impl(
+    const size_t _Size, void* const _String, void* const _Context, const _Stacktrace_string_fill_callback _Callback) {
+    if (_Callback) {
+        static_cast<string*>(_String)->resize_and_overwrite(_Size,
+            [_Callback, _Context](char* _Data, size_t _Size) noexcept { return _Callback(_Data, _Size, _Context); });
+        return static_cast<string*>(_String)->size();
+    } else {
+        return static_cast<string*>(_String)->capacity();
+    }
+}
+
+class stacktrace_entry {
+public:
+    using native_handle_type = void*;
+
+    constexpr stacktrace_entry() noexcept                        = default;
+    constexpr stacktrace_entry(const stacktrace_entry&) noexcept = default;
+    constexpr stacktrace_entry& operator=(const stacktrace_entry&) noexcept = default;
+
+    ~stacktrace_entry() = default;
+
+    _NODISCARD constexpr native_handle_type native_handle() const noexcept {
+        return _Address;
+    }
+
+    _NODISCARD constexpr explicit operator bool() const noexcept {
+        return _Address != nullptr;
+    }
+
+    _NODISCARD string description() const {
+        string _Result;
+        __std_stacktrace_description(_Address, &_Result, _Stacktrace_string_fill_impl);
+        return _Result;
+    }
+
+    _NODISCARD string source_file() const {
+        string _Result;
+        __std_stacktrace_source_file(_Address, &_Result, _Stacktrace_string_fill_impl);
+        return _Result;
+    }
+
+    _NODISCARD uint_least32_t source_line() const noexcept /* strengthened */ {
+        return __std_stacktrace_source_line(_Address);
+    }
+
+    _NODISCARD_FRIEND constexpr bool operator==(const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+    _NODISCARD_FRIEND constexpr strong_ordering operator<=>(
+        const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+private:
+    void* _Address = nullptr;
+};
+
+template <class _Alloc>
+class basic_stacktrace {

```

Comment:
I think we need `static_assert`s here to make certain that `sizeof(stacktrace_entry) == sizeof(void*)`, and `alignof(stacktrace_entry) == alignof(void*)`, since we `reinterpret_cast<void**>(stacktrace_entry*)`. Additionally, I think a function call `to_voidptr_array` or smth that does that cast would be good, so we only have _one_ `reinterpret_cast` to look at.